### ISSUE TITLE
interleaved mode bug

### ISSUE THREAD

**Issue by jianshu93** [BenLangmead/bowtie2#441]
Dear bowtie2 team, 

For version 2.5.0, there are bugs with interleaved mode but not with R1 and R2 mode, see details below;

Jianshus-MBP-2:test_data jianshuzhao$ time bowtie2 -x all_mag.fasta  -f -1 T4AerOil_sbsmpl5.R1.fa.gz -2 T4AerOil_sbsmpl5.R2.fa.gz  -S try.sam --threads 8
1157425 reads; of these:
  1157425 (100.00%) were paired; of these:
    1012592 (87.49%) aligned concordantly 0 times
    144601 (12.49%) aligned concordantly exactly 1 time
    232 (0.02%) aligned concordantly >1 times
    ----
    1012592 pairs aligned concordantly 0 times; of these:
      18872 (1.86%) aligned discordantly 1 time
    ----
    993720 pairs aligned 0 times concordantly or discordantly; of these:
      1987440 mates make up the pairs; of these:
        1982525 (99.75%) aligned 0 times
        4731 (0.24%) aligned exactly 1 time
        184 (0.01%) aligned >1 times
14.36% overall alignment rate

real	0m6.884s
user	0m51.170s
sys	0m0.767s

Jianshus-MBP-2:test_data jianshuzhao$ time bowtie2 -x all_mag.fasta  -f --interleaved T4AerOil_sbsmpl5.fa  -S try.sam --threads 8
(ERR): bowtie2-align died with signal 11 (SEGV) 

interleaved file was created by seqtk mergepe R1.fa.gz R2.fa.gz.  This bug has been there for at least several versions. Can you please check why, the sam output is identical though.

Thanks,

Jianshu

**Comment by jianshu93** [comment]
By the way, interleaved reads file looks like this:

**Comment by ch4rr0** [comment]
Hello @Jianshu93

Thank you for reporting this issue. I have pushed a change to the `bug_fixes` branch that should resolve this.

``` text
./bowtie2-align-s-debug -x example/index/lambda_virus -f  --interleaved foo.fa
Warning: Running in debug mode.  Please use debug mode only for diagnosing errors, and not for typical use of Bowtie 2.
2 reads; of these:
  2 (100.00%) were paired; of these:
    2 (100.00%) aligned concordantly 0 times
    0 (0.00%) aligned concordantly exactly 1 time
    0 (0.00%) aligned concordantly >1 times
    ----
    2 pairs aligned concordantly 0 times; of these:
      0 (0.00%) aligned discordantly 1 time
    ----
    2 pairs aligned 0 times concordantly or discordantly; of these:
      4 mates make up the pairs; of these:
        4 (100.00%) aligned 0 times
        0 (0.00%) aligned exactly 1 time
        0 (0.00%) aligned >1 times
0.00% overall alignment rate
@HD	VN:1.5	SO:unsorted	GO:query
@SQ	SN:gi|9626243|ref|NC_001416.1|	LN:48502
@PG	ID:bowtie2	PN:bowtie2	VN:2.5.1	CL:"/home/bowtie2/bowtie2-align-s-debug -x example/index/lambda_virus -f --interleaved foo.fa"
readPair_20168658_read1	77	*	0	0	*	*	0	0	GGGCAGCAGATGGAGGTCGATGGCCCATATCTCTATCCGCCGCTGATCAGTTCGGTGGCGCCACGCAGCGCGGCGATGGATGCGGGGTTTCGCCCCGGCGATGTGATCGTTGCCGTCGATGGCGACCCGATCTTTGCCTTCGACCAGTTGA	IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII	YT:Z:UP
readPair_20168658_read2	141	*	0	0	*	*	0	0	CGTCGGTGCGCCGCGGCGCCAGCGCGAATTGCAGGTTTTCACCCTCACGCCAGACCGACACTTCCAGAACCCGTCCATCCGACGCCTCGACCCGGCTTTTCAACTGGTCGAAGGCAAAGATCGGGTCGCCATCGACGGCAACGATCACAT	IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII	YT:Z:UP
readPair_3900954_read1	77	*	0	0	*	*	0	0	AGGTGGTAGTGATTTTTTCGGGCTCGAGTTCGAGTACCAGGTAACCATTGTTTGTGGTGTTGCTGTAATGCACGTGCGGGTTGGTCAGTGGCGCGAGTTCCTGGCCAGGGATGCCAGGCATGGAAATAGCCGGGGTGGCGAATTCCACAG	IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII	YT:Z:UP
readPair_3900954_read2	141	*	0	0	*	*	0	0	ACATCATTGAGGGCCAGCAACTGGATAACGTCGTCATTCTCACCGGTGATGTACATTCGTCCTGGGCCATAGATGTCCCGCGTGACCCCTACGACCCGA	IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII	YT:Z:UP
```

**Comment by jianshu93** [comment]
Hello @ch4rr0 ,

I can confirm that it is fixed:

Jianshus-MBP-2:test_data jianshuzhao$ time bowtie2 -x all_mag.fasta  -f --interleaved T4AerOil_sbsmpl5.fa  -S try.sam --threads 8
1157425 reads; of these:
  1157425 (100.00%) were paired; of these:
    1011517 (87.39%) aligned concordantly 0 times
    145616 (12.58%) aligned concordantly exactly 1 time
    292 (0.03%) aligned concordantly >1 times
    ----
    1011517 pairs aligned concordantly 0 times; of these:
      17903 (1.77%) aligned discordantly 1 time
    ----
    993614 pairs aligned 0 times concordantly or discordantly; of these:
      1987228 mates make up the pairs; of these:
        1982475 (99.76%) aligned 0 times
        4640 (0.23%) aligned exactly 1 time
        113 (0.01%) aligned >1 times
14.36% overall alignment rate

real	0m5.379s
user	0m40.471s
sys	0m0.789s


Please update bio conda channel soon if possible. This mode was used by a many teams or pipelines. 


Thanks,

Jianshu

**Comment by ch4rr0** [comment]
This has been fixed in v2.5.2. Thanks for reporting.
